content/Interfaces/Interfaces.q.md

```diff
@@ -0,0 +1,2450 @@
+            $this->age = $newAge;
+        }
+    }
+
+    public function getAge()
+    {
+        return $this->age;
+    }
+
+    public function getName()
+    {
+        return $this->name;
+    }
+
+    private function checkAge($age)
+    {
+        if ($age > 0) {
+            return true;
+        }
+        return false;
+    }
+
+    public function species()
+    {
+        return "human";
+    }
+	
+    public function stage()
+    {
+        return "adolescent";
+    }
+	
+    public function display()
+    {
+        echo $this->name . " is an " . $this->stage() . " " . $this->species() . " being.";  
+    }
+}
+
+$pObject = new Person("Anna", 12);
+$pObject->display();
+?>
+```
+
+
+:::
+
+/// type=REPL, readonly=true, filename=[LifeCycle.php,Person.php]
+
+```php
+<?php
+interfaces LifeCycle
+{
+  public function species();
+  public function stage();
+}
+?>
+```
+
+```php
+<?php
+class Person implements LifeCycle
+{
+    protected $name;
+    private $age;
+	
+    public function __construct($name, $age)
+    {
+        $this->name = $name;
+        if ($this->checkAge($age)) {
+            $this->age = $age;
+        }
+    }
+    
+    public function setAge($newAge)
+    {
+        if ($this->checkAge($newAge)) {
+            $this->age = $newAge;
+        }
+    }
+
+    public function getAge()
+    {
+        return $this->age;
+    }
+
+    public function getName()
+    {
+        return $this->name;
+    }
+
+    private function checkAge($age)
+    {
+        if ($age > 0) {
+            return true;
+        }
+        return false;
+    }
+
+    public function species()
+    {
+        return "human";
+    }
+	
+    public function stage()
+    {
+        return "adolescent";
+    }
+	
+    public function display()
+    {
+        echo $this->name . " is an " . $this->stage() . " " . $this->species() . " being.";  
+    }
+}
+
+$pObject = new Person("Anna", 12);
+$pObject->display();
+?>
+```
+/// type=SS, answer=[1]
+
+Execute the program. What is the error message?
+
+ - syntax error, unexpected `'LifeCycle'` (T_STRING) on line number 2
+
+ - Interface function `LifeCycle::species()` cannot contain body on line number 4
+
+ - syntax error, unexpected `'{'`, expecting function (T_FUNCTION) on line number 40
+
+ - syntax error, unexpected `'implement'` (T_STRING), expecting `'{'` on line number 2
+
+ - Class Student contains `1` abstract method and must therefore be declared abstract or implement the remaining methods (LifeCycle::stage) on line number 2
+
+
+/// type=MS, answer=[1,3]
+
+Which statements correctly describe the error?
+
+ - On line 2 of `LifeCycle.php`, the declaration `interfaces LifeCycle` is invalid.
+
+ - There is no semicolon `;` at the end of the statement on line 2 of `LifeCycle.php`.
+
+ - The `interface` keyword is misspelled as `interfaces` on line 2 of `LifeCycle.php`.
+
+ - There is no open curly `{` after `interfaces LifeCycle` on line 2 of `LifeCycle.php`.
```

Comment:
Suggestion on: `open curly {`
What if: `open bracket {`

Reply:
My mistake, thank you.